pkg/core/resources/apis/system/global_secret.go

```diff
@@ -0,0 +1,91 @@
+package system
+
+import (
+	"errors"
+
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/registry"
+)
+
+const (
+	GlobalSecretType model.ResourceType = "GlobalSecret"
+)
+
+var _ model.Resource = &GlobalSecretResource{}
+
+type GlobalSecretResource struct {
+	Meta model.ResourceMeta
+	Spec *system_proto.Secret
+}
+
+func NewGlobalSecretResource() *GlobalSecretResource {
+	return &GlobalSecretResource{
+		Spec: &system_proto.Secret{},
+	}
+}
+
+func (t *GlobalSecretResource) GetType() model.ResourceType {
+	return GlobalSecretType
+}
+func (t *GlobalSecretResource) GetMeta() model.ResourceMeta {
+	return t.Meta
+}
+func (t *GlobalSecretResource) SetMeta(m model.ResourceMeta) {
+	t.Meta = m
+}
+func (t *GlobalSecretResource) GetSpec() model.ResourceSpec {
+	return t.Spec
+}
+func (t *GlobalSecretResource) SetSpec(spec model.ResourceSpec) error {
+	value, ok := spec.(*system_proto.Secret)
+	if !ok {
+		return errors.New("invalid type of spec")
+	} else {
+		t.Spec = value
+		return nil
+	}
+}
+func (t *GlobalSecretResource) Validate() error {
+	return nil
+}
+func (t *GlobalSecretResource) Scope() model.ResourceScope {
+	return model.ScopeGlobal
+}
+
+var _ model.ResourceList = &GlobalSecretResourceList{}
+
+type GlobalSecretResourceList struct {
+	Items      []*GlobalSecretResource
+	Pagination model.Pagination
+}
+
+func (l *GlobalSecretResourceList) GetItems() []model.Resource {
+	res := make([]model.Resource, len(l.Items))
+	for i, elem := range l.Items {
+		res[i] = elem
+	}
+	return res
+}
+func (l *GlobalSecretResourceList) GetItemType() model.ResourceType {
+	return GlobalSecretType
+}
+func (l *GlobalSecretResourceList) NewItem() model.Resource {
+	return NewGlobalSecretResource()
+}
+func (l *GlobalSecretResourceList) AddItem(r model.Resource) error {
+	if trr, ok := r.(*GlobalSecretResource); ok {
+		l.Items = append(l.Items, trr)
+		return nil
+	} else {
+		return model.ErrorInvalidItemType((*GlobalSecretResource)(nil), r)
+	}
```

Comment:
maybe:
```suggestion
	if trr, ok := r.(*GlobalSecretResource); ok {
		l.Items = append(l.Items, trr)
		return nil
	}

	return model.ErrorInvalidItemType((*GlobalSecretResource)(nil), r)
```

Reply:
Those two nits are valid points, but I want this file to be consistent with other Resource files, so I'd prefer to leave it as is